File: app/javascript/channels/notifications_channel.js
import consumer from "./consumer"

let MyApp = {
  notificationReceived: function(data) {
    console.log(data);
  }
}

consumer.subscriptions.create("NotificationsChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    this.typing = this.typing.bind(this);
    document.addEventListener('keyup', this.typing)
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    MyApp.notificationReceived(data);
  },

  typing() {
    // Invoke #isTyping on notifications_channel.rb
    this.perform('typing');
  }
});

File: app/channels/notifications_channel.rb
class NotificationsChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'all_notifications'
    ActionCable.server.broadcast('all_notifications', {
      user: current_user,
      message: 'has joined'
    })
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def typing
    ActionCable.server.broadcast('all_notifications', {
      user: current_user,
      message: "is typing"
    })
  end
end
